rivertest/rivertest.go

```diff
@@ -157,14 +166,117 @@ func requireInsertedErr[TDriver riverdriver.Driver[TTx], TTx any, TArgs river.Jo
 	}
 
 	if opts != nil {
-		if !compareJobToInsertOpts(t, jobRow, *opts, -1) {
+		if compareJobToInsertOpts(t, jobRow, opts, -1, false) == compareResCheckFailed {
 			return nil, nil //nolint:nilnil
 		}
 	}
 
 	return &river.Job[TArgs]{JobRow: jobRow, Args: actualArgs}, nil
 }
 
+// RequireNotInserted is a test helper that verifies that a job of the given
+// kind was not inserted for work, failing the test if one was.
+//
+//	job := RequireNotInserted(ctx, t, riverpgxv5.New(dbPool), &Job1Args{}, nil)
+//
+// This variant takes a driver that wraps a database pool. See also
+// RequireNotInsertedTx which takes a transaction.
+//
+// A RequireInsertedOpts struct can be provided as the last argument, and if it
+// is, its properties (e.g. max attempts, priority, queue name) will act as
+// requirements on a found row. If any fields are set, then the test will fail
+// if a job is found that matches any of them (unlike options to RequireInserted
+// which behave like an AND, these are an OR). So for example, if options
+// specify `Priority: 3`, and a joke of the same kind was inserted, but it was
+// `Priority: 2`, RequireNotInserted will not fail. If the inserted job was
+// `Priority: 2` (therefore matching the job), RequireNotInserted does fail.
+func RequireNotInserted[TDriver riverdriver.Driver[TTx], TTx any, TArgs river.JobArgs](ctx context.Context, tb testing.TB, driver TDriver, expectedJob TArgs, opts *RequireInsertedOpts) {
```

Comment:
Okay, so I went back and forth a lot on what `opts` this function should accept.

You can see here that it takes `RequireInsertedOpts`, which is the same as `RequireInserted`. Initially, this felt a little wrong to me, but after implementing it as a different struct, having a totally duplicated set of fields on a different struct just felt wasteful.

I then thought about hacking it a bit with something like:

``` go
type RequireNotInsertedOpts RequireInsertedOpts
```

Which would let us futureproof a bit by changing the type definition to a full struct in the future in case we needed to. But it has the unfortunate downside of being worse for IDE mouseover definitions of the struct, because the properties of `RequireInsertedOpts` aren't directly visible.

Lastly, I considered just not giving this function an `opts` at all. The purpose of the opts is so you could do something like:

``` go
RequireNotInserted(..., &RequireInsertedOpts{Queue: "high-priority"})
```

To verify that a job of that type _did not_ go the high property queue. So if a job the kind went to the default queue, it's okay, but it fails if it went to high priority.

However, one could also imagine asserting on this affirmatively instead by inverting. Something like:

``` go
RequireInserted(..., &RequireInsertedOpts{Queue: "default"})
```

Which might be clearer than the use of `RequireNotInserted` anyway.

That said, if we decide not to give `RequireNotInserted` options, then realize later that we need them for something, it's a breaking change to add them in, which would be nice to avoid.

Any immediate thoughts on all that?

Reply:
See my above comment, I think the behavior on this might be backwards.

But as far as the type thing, I would definitely avoid the aliasing option because it results in poorer usability. If we define these types right next to each other, it wouldn't be that hard to keep them lined up.  Or alternatively if we can get away with just using the same type in both places, that's even better.